services/evm/executor.go

```diff
@@ -0,0 +1,196 @@
+package evm
+
+import (
+	"fmt"
+
+	"github.com/onflow/atree"
+	"github.com/onflow/flow-go/fvm/evm"
+	"github.com/onflow/flow-go/fvm/evm/emulator"
+	"github.com/onflow/flow-go/fvm/evm/emulator/state"
+	"github.com/onflow/flow-go/fvm/evm/precompiles"
+	"github.com/onflow/flow-go/fvm/evm/types"
+	flowGo "github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/go-ethereum/common"
+	gethTypes "github.com/onflow/go-ethereum/core/types"
+	"github.com/onflow/go-ethereum/eth/tracers"
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-evm-gateway/models"
+	"github.com/onflow/flow-evm-gateway/storage"
+)
+
+type BlockExecutor struct {
+	types.StateDB // todo change to types.ReadOnlyView
+	emulator      types.Emulator
+	chainID       flowGo.ChainID
+	block         *models.Block
+	blocks        storage.BlockIndexer
+	logger        zerolog.Logger
+	receipts      storage.ReceiptIndexer
+
+	// block dynamic data
+	txIndex uint
+	gasUsed uint64
+}
+
+func NewBlockExecutor(
+	block *models.Block,
+	ledger atree.Ledger,
+	chainID flowGo.ChainID,
+	blocks storage.BlockIndexer,
+	receipts storage.ReceiptIndexer,
+	logger zerolog.Logger,
+) (*BlockExecutor, error) {
+	logger = logger.With().Str("component", "state-execution").Logger()
+	storageAddress := evm.StorageAccountAddress(chainID)
+
+	stateDB, err := state.NewStateDB(ledger, storageAddress)
+	if err != nil {
+		return nil, err
+	}
+
+	return &BlockExecutor{
+		emulator: emulator.NewEmulator(ledger, storageAddress),
+		StateDB:  stateDB,
+		chainID:  chainID,
+		block:    block,
+		blocks:   blocks,
+		receipts: receipts,
+		logger:   logger,
+	}, nil
+}
+
+func (s *BlockExecutor) Run(
```

Comment:
This is currently not used. Lets maybe save it for a different PR.

Reply:
Removed in https://github.com/onflow/flow-evm-gateway/pull/605/commits/cc3d1157efdbc63d2457dbc3fe7c42a2c53b4617